lib/metadata/index.js

```diff
@@ -125,6 +125,11 @@ Metadata.prototype.buildTokens = function (hosts) {
   this.primaryReplicas = primaryReplicas;
   //All the tokens in ring order
   this.ring = allSorted;
+  //Compute string versions as it's potentially expensive and frequently reused later
+  this.ringTokensAsStrings = [];
+  for (var i = 0; i < allSorted; i++) {
```

Comment:
It's only covered in the integration tests (`test/short/load-balancing-tests.js`). If you want to run them locally you must install [ccm](https://github.com/pcmanus/ccm) or you can let CI do that for you :)

We could include a specific unit test for it on `test/unit/metadata-tests.js` for `buildTokens()` making sure that the necessary values are populated, ie: `primaryReplicas`, `ring`, `ringTokensAsStrings`, ...
Do you think you can include it as part of this patch? otherwise, I could do it.